### PR TITLE
clean unused test dependencies

### PR DESCRIPTION
**Pull request by bnavigator**
Since the drop of Python 2, `six`, `mock` and `pbr` are not used anywhere. Remove them from the test dependencies.[ It already confused a packager](https://lists.opensuse.org/archives/list/python@lists.opensuse.org/message/XOCQ5ZASO47IZLFAOB5VFTS2C4OLLKOM/).

### CODE REVIEW

**Comment by joachimmetz** [comment]
Thx for flagging, some of these files are generated, I'll have a look to update the generator first. yeah packagers do strange things, since there is no need to package the tests.

**Comment by bnavigator** [comment]
No need to package. But use them in order to make sure that the package works.

**Comment by joachimmetz** [comment]
> No need to package. But use them in order to make sure that the package works.

these are unit tests not deployment tests, so they give you no guarantee about this

**Comment by bnavigator** [comment]
Of course, but it is better than nothing.

https://en.opensuse.org/openSUSE:Packaging_Python#Running_tests

**Comment by joachimmetz** [comment]
But you're running these during packing, not on the packaged results, so they are basically testing if you can run the unit test on the build systems

**Comment by joachimmetz** [comment]
Changed the generator and corresponding changes in https://github.com/log2timeline/dfdatetime/pull/247

File: run_tests.py [hunk 1]
```diff
@@ -2,8 +2,6 @@
 # -*- coding: utf-8 -*-
 """Script to run the tests."""
 
-from __future__ import print_function
```
Comment (joachimmetz) [inline run_tests.py:5]: this is needed if someone runs setup.py with python2

File: setup.py [hunk 2]
```diff
@@ -214,5 +214,4 @@ def parse_requirements_from_file(path):
             'ACKNOWLEDGEMENTS', 'AUTHORS', 'LICENSE', 'README']),
     ],
     install_requires=parse_requirements_from_file('requirements.txt'),
-    tests_require=parse_requirements_from_file('test_requirements.txt'),
```
Comment (joachimmetz) [inline setup.py:217]: this is generated I opt to keep this

File: config/appveyor/install.ps1 [hunk 3]
```diff
@@ -18,6 +15,6 @@ New-Item -ItemType "directory" -Name "dependencies"
 
 $env:PYTHONPATH = "..\l2tdevtools"
 
-$Output = Invoke-Expression -Command "& '${env:PYTHON}\python.exe' ..\l2tdevtools\tools\update.py --download-directory dependencies --machine-type ${env:MACHINE_TYPE} --msi-targetdir ${env:PYTHON} --track ${env:L2TBINARIES_TRACK} ${Dependencies} 2>&1"
```
Comment (joachimmetz) [inline config/appveyor/install.ps1:21]: this is generated I opt to keep this

File: run_tests.py (hunk 1 above)
Comment (bnavigator) [reply run_tests.py:5]: Python 2 is dead. You only support Python >= 3.7

Comment (joachimmetz) [reply run_tests.py:5]: correct python 2 is no longer maintained/supported, but that does not mean someone is still able to run it

**Comment by bnavigator** [comment]
> But you're running these during packaging, not on the packaged results

That is not correct. ["We" run the tests](https://build.opensuse.org/package/view_file/security:forensics/python-dfdatetime/python-dfdatetime.spec?expand=1) on the [packaged results](https://github.com/openSUSE/python-rpm-macros#unit-testing), but that is not what this PR or the linked mailinginlist post is about.

File: run_tests.py (hunk 1 above)
Comment (bnavigator) [reply run_tests.py:5]: https://github.com/log2timeline/dfdatetime/blob/de021f3d198c47bd6d616dc68bd59f83a6158961/setup.py#L25-L28

Comment (bnavigator) [reply run_tests.py:5]: It will fail at the f-string.

Comment (joachimmetz) [reply run_tests.py:5]: Good point I'll have a look at a later point at this.

**Comment by joachimmetz** [comment]
> That is not correct. ["We" run the tests](https://build.opensuse.org/package/view_file/security:forensics/python-dfdatetime/python-dfdatetime.spec?expand=1) on the [packaged results](https://github.com/openSUSE/python-rpm-macros#unit-testing), but that is not what this PR or the linked mailinginlist post is about.

Correct this is outside the scope of this PR but isn't `%check` run before the rpm is created? so any changes tests could make to the "installed" (pre package) files are included in the package?

**Comment by bnavigator** [comment]
Yes, but the checks are run on the `%buildroot` which is where all the files have been installed by `%install` and are packaged from in a later state. Again, this has nothing to do with this PR. The commands in the specfile `%check` section do not run `run_tests.py` but use `unittest` directly.

> so any changes tests could make to the "installed" (pre packaged) files are included in the package (rpm)?

That would be the case. Although changes to the `%buildroot` are only allowed in `%install`, not `%check`. `PYTHONDONTWRITEBYTECODE=1` is set and who changes source files in unit tests? That sounds dumb.

**Comment by joachimmetz** [comment]
> who changes source files in unit tests?

Consider someone with malicious intent.

**Comment by bnavigator** [comment]
Well that's on the packager. I don't see much difference in the source code itself and the source code for tests. Malicious code could be in both.

**Comment by joachimmetz** [comment]
true, but it is an additional attack surface or surface for additional errors, but agree that is the responsibility of the  packager/distribution

**Pull request closed without merging**